svc-mesh/service-mesh.md

```diff
@@ -0,0 +1,251 @@
+file. We are taking their [emojivoto](https://conduit.io/getting-started/index.html)
+sample as example
+```
+$ wget https://raw.githubusercontent.com/runconduit/conduit-examples/master/emojivoto/emojivoto.yml
+$ conduit inject emojivoto.yml > emojivoto-injected.yaml
+```
+
+## Run your services with Kata
+
+Now that you have your service deployment injected with the appropriate
+sidecar containers, you need to manually edit your deployment to make it
+work with Kata.
+
+### Lower privileges
+
+In Kubernetes, the __init__ container is often `privileged` as it needs to
+setup the environment, which often needs some root privileges. But in case
+of those service meshes, all they need is the `NET_ADMIN` capability so that
+they can modify the underlying network rules. Conduit, by default, does not
+use `privileged` container, but Istio does.
+
+For this reason, if you are using Istio, you will need to switch all
+containers with `privileged: true` to `privileged: false`.
+
+### Add annotations
+
+This last section is the common part which does not make any difference
+between Istio and Conduit because it is about which CRI implementation
+you are using.
+
+For both CRI-O and CRI-containerd, you will have to add an annotation
+indicating the workload for this deployment is not _trusted_, which will
+trigger `kata-runtime` to be called instead of `runc`.
+
+__CRI-O__
+
+Add the following annotation for CRI-O
+```yaml
+io.kubernetes.cri-o.TrustedSandbox: "false"
+```
+Here is what your YAML could look like 
+
+```yaml
+...
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  creationTimestamp: null
+  name: details-v1
+spec:
+  replicas: 1
+  strategy: {}
+  template:
+    metadata:
+      annotations:
+        io.kubernetes.cri-o.TrustedSandbox: "false"
+        sidecar.istio.io/status: '{"version":"55c9e544b52e1d4e45d18a58d0b34ba4b72531e45fb6d1572c77191422556ffc","initContainers":["istio-init"],"containers":["istio-proxy"],"volumes":["istio-envoy","istio-certs"],"imagePullSecrets":null}'
+      creationTimestamp: null
+      labels:
+        app: details
+        version: v1
+...
+```
+
+__CRI-containerd__
+
+Add the following annotation for CRI-containerd
+```yaml
+io.kubernetes.cri.untrusted-workload: "true"
+```
+Here is what your YAML could look like 
+
+```yaml
+...
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  creationTimestamp: null
+  name: details-v1
+spec:
+  replicas: 1
+  strategy: {}
+  template:
+    metadata:
+      annotations:
+        io.kubernetes.cri.untrusted-workload: "true"
+        sidecar.istio.io/status: '{"version":"55c9e544b52e1d4e45d18a58d0b34ba4b72531e45fb6d1572c77191422556ffc","initContainers":["istio-init"],"containers":["istio-proxy"],"volumes":["istio-envoy","istio-certs"],"imagePullSecrets":null}'
+      creationTimestamp: null
+      labels:
+        app: details
+        version: v1
+...
+```
+
+### Deploy !
+
+You only need to deploy your application at this point, simply using
```

Comment:
Deploy your application by using the following:

Reply:
done